src/Data/Bool/Linear.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+-- | This module provides linear functions on the standard 'Bool' type.
+module Data.Bool.Linear
+  ( -- * The Boolean type
+    Bool(..)
+    -- * Operators
+  , (&&)
+  , (||)
+  , not
+  )
+  where
+
+import Prelude (Bool(..))
+
+-- | True iff both are True.
+(&&) :: Bool #-> Bool #-> Bool
```

Comment:
I'm wondering out loud whether we should take the opportunity to make a boolean algebra type class to abstract over these? Maybe not, it many not be useful often enough to be featured in a standard library.

Reply:
Do this functions deserve a word about being strict? (unlike their non-linear variants)

Reply:
This is a good point. Well-spotted.

Reply:
Yes linearity forces them to be strict. That's interesting -- so in some sense giving linear types to functions in `base` that are linear does impact performance. (So a non-linear function using the linear `&&` non-linearly might have a different performance. So linear types don't impact the performance of a function that's already written, but they do force constraints on the implementation which might impact performance. This is an interesting observation.)